packages/mui-material/src/Menu/Menu.js

```diff
@@ -172,13 +180,34 @@ const Menu = React.forwardRef(function Menu(inProps, ref) {
     className: [classes.root, className],
   });
 
-  const paperSlotProps = useSlotProps({
-    elementType: PaperSlot,
-    externalSlotProps: paperExternalSlotProps,
-    ownerState,
+  const [PaperSlot, paperSlotProps] = useSlot('paper', {
     className: classes.paper,
+    elementType: MenuPaper,
+    externalForwardedProps,
+    shouldForwardComponentProp: true,
+    ownerState,
+  });
+
+  const [ListSlot, listSlotProps] = useSlot('list', {
+    className: clsx(classes.list, MenuListProps.className),
+    elementType: MenuMenuList,
+    shouldForwardComponentProp: true,
+    externalForwardedProps,
+    getSlotProps: (handlers) => ({
+      ...handlers,
+      onKeyDown: (event) => {
+        handleListKeyDown(event);
+        handlers.onKeyDown?.(event);
+      },
+    }),
+    ownerState,
   });
 
+  const resolvedTransitionProps =
+    typeof externalForwardedProps.slotProps.transition === 'function'
+      ? externalForwardedProps.slotProps.transition(ownerState)
+      : externalForwardedProps.slotProps.transition;
```

Comment:
Should we use `mergeSlotProps` here?

Reply:
`mergeSlotProps` does not concat event handlers, if users provide `onEntering` it will replace the `handleEntering` of the Menu. That's why we need to check for typeof function here so that we can:

```js
  onEntering: (...args) => {
    handleEntering(...args); // menu logic
    resolvedTransitionProps?.onEntering?.(...args); // user logic
  },
```

I don't want to go that far to make `mergeSlotProps` handle event handlers concatenation because it might effect the performance because the logic will need to loop all slotProps to identify which prop is an event handler (starting with `on*`).